jetty9-compat-base/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java

```diff
@@ -195,6 +187,28 @@ protected void doStart() throws Exception {
     }
 
     super.doStart();
+
+    // Look for a datastore service within the context
+    ClassLoader orig = Thread.currentThread().getContextClassLoader();
```

Comment:
Reflection is necessary here because we are pushing the webapps to providing their own implementation.  While we do have an implementation on the container classpath, it is for our session manager and is a different instance than will be used by the webapp.    So in order to deal with the transactions of the webapp, we need to either use reflection or move the code to a filter that we can load from the webapp classpath.